lib/rules/no-compare-neg-zero.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview The rule should warn against code that tries to compare against -0.
+ * @author no-compare-neg-zero
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow compare against -0.",
```

Comment:
Small nitpick: "disallow comparing against -0" or "disallow comparisons against -0", and I think the period (.) should probably be removed as well.